### PR TITLE
Add Microsoft UI fabric

### LINKED ISSUE #22: Sort the data, according to the specified fields
Sort the data retrieved from the API into the relevant fields on the User Interface

Related to user story #12

### PR DESCRIPTION
**Pull request by ElinorW**
## Description

Addition of the Microsoft UI Fabric.
Displaying the data in a Details List instead of a table.
Added sorting functionality to columns.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### Closing issue
Fixes #22

### CODE REVIEW

File: SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                        onChange={this._onFilterName}
+                  styles={{ root: { maxWidth: '300px' } }}
+                    />
+                    </Sticky>
+                    {announcedMessage ? <Announced message={announcedMessage} /> : undefined}
+                <MarqueeSelection selection={this._selection} data-is-scrollable={true}>
+                <DetailsList
+                    items={items}
+                    columns={columns}
+                    selectionMode={SelectionMode.multiple}
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx]: Lets change this to `SelectionMode.none` for now.

File: SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    announcedMessage?: string;
+}
+
+export default class ListData extends React.Component<{}, IListDataState> {
+    private _selection: Selection;
+    private _allItems: IListDataItem[];
+
+    constructor(props: {}) {
+        super(props);
+
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx]: Check out https://github.com/OfficeDev/office-ui-fabric-react/wiki/Using-icons.

You the missing icons will show once you this line here
```js 
initializeIcons(/* optional base url */);
```

File: SamplesDashboard/SamplesDashboard/ClientApp/src/components/Home.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ class Home extends Component {
                 <br/>
                 <h2> List of Samples </h2>
                 <br />
-                <DataTable  />
+                <ListData />
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/ClientApp/src/components/Home.js]: Lets call this "SampleList"

File: SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        this._selection = new Selection({
+        onSelectionChanged: () => this.setState({ selectionDetails: this._getSelectionDetails() })
+        });
+        // Populate with items for demos.
+        this._allItems = [];
+        const columns: IColumn[] = [
+            { key: 'column1', name: 'Name', fieldName: 'name', minWidth: 200, maxWidth: 300, isRowHeader: true, isResizable: true, isSorted: true, isSortedDescending: false, onColumnClick: this._onColumnClick },
```
Comment (thewahome) [inline SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx]: Rename the columns to something more readable.. You can repeat the fieldName value in this case

File: SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+      case 0:
+        return 'No items selected';
+      case 1:
+        return '1 item selected: ' + (this._selection.getSelection()[0] as IListDataItem).name;
+      default:
+        return `${selectionCount} items selected`;
+    }
+  }
+
+    private _onFilterName = (ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, text: string): void => {
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx]: You can fix build issue with
```suggestion
        private _onFilterName = (ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, text?: string | undefined): void => {
```

File: SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx (hunk 4 above)
Comment (ElinorW) [reply SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx]: Do you mean the column key or the column name??

File: SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx (hunk 5 above)
Comment (ElinorW) [reply SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx]: Oh okay thank you for this

**Pull request merged**